RFCs/codepush-for-mobile-releases.md

```diff
@@ -0,0 +1,75 @@
+
+Now, on to the main part of this rfc. Here is how I want this to work (more info on the tech plan).
+
+### for PRs
+
+Codepush allows for multiple "deployments", as they call them. Imagine each deployment is like a git branch, and each update is like a commit on that branch. So if your app on your device is on deployment `cool-deploy`, then your app will update every time there is an update on the `cool-deploy` deployment.
+
+Deployments are cheap to make and easy to delete. So the idea here is this:
+
+- For each PR, we make a deploy. For PR #123 for example, we make a deployment called `PR-123`. This happens on CI.
+- On our app, in the admin menu, we will have a new list showing, with all the open PRs, their number and title. Each of those will be tappable.
+- Once you tap on one, the app will change to the deployment of the tapped PR.
+- Once this happens, we will force the app to make a check for the newest bundle on the deployment, and get it, and start using it.
+- Behold, at this point we are using code from a PR, that is not merged yet, and we can run manual tests and do QA on it.
+
+Even more importantly, now anyone with the beta app can do this! No need to try to compile the app so you can test something. Designers can immediately run a PR and see the progress. Same for PMs. Same for any engineer that doesn't run eigen often, and doesn't want to spend half an hour setting up and building. Same for the engineers that have eigen ready to goon their machine, because it's still easier to just shake the app, tap 3 times, and have the newest js bundle of a PR, rather than stash, change branches, wait for js to build locally etc.
+
+
+### for releases
+
+Any release that includes changes on native code or anything related, like fonts or plist files etc, will still go through the regular appstores process. This is the only way to get these updates to the users' devices, using native updates.
+
+Any release that includes only js/ts file updates, even npm packages (if they don't include native code), they can do either through the regular appstores release process, or codepush. My proposal here would be to have codepush as the default, and only fallback to appstores release process every few releases, or if we have an important change that Apple or Google cares to see. We can keep our cadence as is for now, and re-evaluate later, as we have all the processes in place. But if we want to be very conservative, I would say we can go for a "1 native, 1 codepush" release schedule, unless there are changes that require a native one of course.
```

Comment:
I am not sure if we should replace the already existing release process with code push 100% 🤔 

Love the idea of seamless releases that bypass the store checks but not sure if we should do it 100% of the time.

Reply:
it could be a start, and something we try. definitely not replacing everything with codepush just yet.

Reply:
I agree. I think we should start by reserving it for hotfixes while we evaluate the potential for becoming more CD, and think _very carefully_ about how we would need to change our processes to support that.